addons/point_of_sale/static/src/app/models/related_models.js

```diff
@@ -713,6 +714,15 @@ export function createRelatedModels(modelDefs, modelClasses = {}, indexes = {})
             const fields = getFields(model);
             for (const rawRec of rawRecords) {
                 const recorded = records[model][rawRec.id];
+                // Check if there are any missing fields for this record
+                const key = `${model}_${rawRec.id}`;
+                if (missingFields[key]) {
+                    for (const [record, field] of missingFields[key]) {
+                        // Connect the `recorded` to the missing `field` in `record`
+                        connect(field, record, recorded);
+                    }
+                    delete missingFields[key];
+                }
```

Comment:
Don't do this here. Put this logic in `loadMissingRecords`.
The idea is that `loadData` should be focusing on putting the given records together and collecting the missing links.
And `loadMissingRecords` is then responsible on fetching the missing data and performing the linking.
Kindly check the commit I added.